### PR TITLE
Tree: Add debug node formatting for browser and NodeJS

### PR DESCRIPTION
**Pull request by CraigMacomber**
## Description

Implement browser and NodeJS specific debug formatters so TreeNodes show their actual type and content instead of just "Proxy" and some junk.

This should be considered an early MVP: the experience isn't great but its an improvement and a starting place to build something nicer.

## Reviewer Guidance

The review process is outlined on [this wiki page](https://github.com/microsoft/FluidFramework/wiki/PR-Guidelines#guidelines).

Probably actually debug a test in node) and example (in browser) to check out the DevX.

Note when testing in the browser, you will need to enable custom formatters in the dev tools.

### CODE REVIEW

File: .changeset/silly-planets-wait.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+"@fluidframework/tree": minor
+---
+
+Add debug visualizers for NodeJS and browsers for TreeNodes
```
Comment (tylerbutler) [inline .changeset/silly-planets-wait.md:5]:
```suggestion
New feature: Debug visualizers for TreeNodes in NodeJS and browsers
```

File: .changeset/silly-planets-wait.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+
+TreeNodes now have custom debug visualizers which should improve the debug experience in NodeJS and in browsers if custom formatters are enabled in the developer tools.
```
Comment (tylerbutler) [inline .changeset/silly-planets-wait.md:7]:
```suggestion
TreeNodes now have custom debug visualizers to improve the debug experience in NodeJS and in browsers. Note that custom formatters must be enabled in the developer tools for these visualizers to be used.
```

In this case are the devtools the browser devtools or the Fluid devtools?

**Review by tylerbutler — APPROVED** [review]
Minor changeset notes - not critical to accept before merge because I can clean it up in #21897 if needed.

File: .changeset/silly-planets-wait.md (hunk 2 above)
Comment (CraigMacomber) [reply .changeset/silly-planets-wait.md:7]: browser.

```suggestion
TreeNodes now have custom debug visualizers to improve the debug experience in NodeJS and in browsers. Note that custom formatters must be enabled in the browser developer tools for that visualizer to be used.
```

File: .changeset/silly-planets-wait.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+---
+"@fluidframework/tree": minor
```
Comment (CraigMacomber) [inline .changeset/silly-planets-wait.md:2]:
```suggestion
"fluid-framework": minor
"@fluidframework/tree": minor
```

File: packages/dds/tree/src/simple-tree/types.ts [hunk 4]
```diff
@@ -323,3 +324,157 @@ export function toFlexTreeNode(node: InternalTreeNode): FlexTreeNode {
...
+	body(object, config): DevtoolsFormatter.Item {
+		const children: DevtoolsFormatter.Item[] = [];
+		for (const [key, value] of Object.entries(object as TreeNode)) {
+			children.push(["li", ["span", `${key}: `], formattedReference(value)]);
+		}
+
+		// TODO:
+		// for array nodes, this isn't great since (at least in FireFox) the list items show up with a prefixed number starting from 1.
+		// This looks messy when followed by the array index.
+		// Find a way to hid the list index.
```
Comment (Josmithr) [inline packages/dds/tree/src/simple-tree/types.ts:415]:
```suggestion
		// Find a way to hide the list index.
```

**Review by Josmithr — APPROVED** [review]
It would be nice if we could add some automated testing for this. Also, in theory, we would want to ensure that whatever formatting we introducing conforms to our accessibility requirements. But since this adds information that was otherwise invisible, is somewhat opt-in, and makes no color formatting changes, it's probably fine. Still, we should follow up on testing / accessibility requirements before we advertise this publicly.

**Comment by CraigMacomber** [comment]
> It would be nice if we could add some automated testing for this. Also, in theory, we would want to ensure that whatever formatting we introducing conforms to our accessibility requirements. But since this adds information that was otherwise invisible, is somewhat opt-in, and makes no color formatting changes, it's probably fine. Still, we should follow up on testing / accessibility requirements before we advertise this publicly.

It will end up in the 2.1 release notes, but I hope is fine. But yes, it should get a lot more care (testing, improved formatting, UI/UX review etc) before we really hilight it as a feature.

**Comment by CraigMacomber** [comment]
CI is not making progress. Doesn't seem to be running ADO pipeline. Neither Closing and opening PR nor rerun all jobs seems to help. Not sure what to do.

**Comment by alexvy86** [comment]
ADO had an outage yesterday, seems to be back now. Closing + reopening should work now, I'd hope.

**Pull request merged**